### PR TITLE
Support specifying custom category in PR tree view

### PR DESCRIPTION
**Pull request by RMacfarlane**
https://github.com/microsoft/vscode-pull-request-github/issues/1106

Experimenting with being able to set custom tree categories using normal GitHub search syntax
queries_setting

This would also open up viewing closed pull requests, need to test that there are no problems with displaying these

### CODE REVIEW

**Comment by kuychaco** [comment]
Oh ya know, I just realized that there might be some UX friction to smooth over before merging this. If the user has the "Use split JSON" option enabled (as shown in the gif above) then it's clear what information users will want to add to their settings.json file. The default is for this setting to be off, in which case it's not quite clear that the user will need to add an array of objects with label and query fields. 

Perhaps we can address this by adding more information in the setting description.

**Comment by RMacfarlane** [comment]
@kuychaco I updated the description of the setting to try to clarify this, let me know if it makes sense to you.

It would be good to have telemetry on this, with that info we could even tweak the default categories. But I'm not sure if this would be OK to log, since it's direct user input.

File: package.json [hunk 1]
```diff
@@ -87,6 +87,38 @@
...
+                "description": "The label to display for the query in the Pull Requests tree"
+              },
+              "query": {
+                "type": "string",
+                "description": "The query used for searching pull requests."
+              }
+            }
+          },
+          "scope": "resource",
+          "markdownDescription": "Specifies what queries should be used in the GitHub Pull Requests tree. Each query has a label that will be shown in the tree and a search query using [GitHub search syntax](https://help.github.com/en/articles/understanding-the-search-syntax). The variable `${user}` can be used to specify the logged in user within a search. By default these queries define the categories \"Waiting For My Review\", \"Assigned To Me\" and \"Created By Me\". If you want to preserve these, make sure they are still in the array when you modify the setting.",
```
Comment (kuychaco) [inline package.json]:
```suggestion
          "markdownDescription": "Specifies what queries should be used in the GitHub Pull Requests tree. Each query object has a `label` that will be shown in the tree and a search `query` using [GitHub search syntax](https://help.github.com/en/articles/understanding-the-search-syntax). The variable `${user}` can be used to specify the logged in user within a search. By default these queries define the categories \"Waiting For My Review\", \"Assigned To Me\" and \"Created By Me\". If you want to preserve these, make sure they are still in the array when you modify the setting.",
```

**Comment by RMacfarlane** [comment]
I'll create a new issue and defer the telemetry for now, so we can double check GDPR. Also, looking at the typings for the telemetry module, right now just logs an action, so it might also need more work to send additional data.

**Pull request merged**